library/core/src/main/java/com/google/android/exoplayer2/scheduler/Requirements.java

```diff
@@ -162,6 +175,34 @@ private boolean isDeviceIdle(Context context) {
         : Util.SDK_INT >= 20 ? !powerManager.isInteractive() : !powerManager.isScreenOn();
   }
 
+  /**
+   * Implementation taken from the the WorkManager source.
+   * @see <a href="https://android.googlesource.com/platform/frameworks/support/+/androidx-master-dev/work/workmanager/src/main/java/androidx/work/impl/constraints/trackers/StorageNotLowTracker.java">StorageNotLowTracker</a>
+   */
+  private boolean isStorageNotLow(Context context) {
+    IntentFilter intentFilter = new IntentFilter();
+    intentFilter.addAction(Intent.ACTION_DEVICE_STORAGE_OK);
+    intentFilter.addAction(Intent.ACTION_DEVICE_STORAGE_LOW);
+    Intent intent = context.registerReceiver(null, intentFilter);
+    if (intent == null || intent.getAction() == null) {
```

Comment:
Do you understand what the purpose of the `intent.getAction() == null` check is here?

Reply:
Ignore this; figured it out!